engine/src/main/java/org/terasology/config/flexible/FlexibleConfigManager.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import org.terasology.engine.SimpleUri;
+
+public interface FlexibleConfigManager {
+    void addFlexibleConfig(SimpleUri flexibleConfigUri, FlexibleConfig fc);
+
+    void removeFlexibleConfig(SimpleUri flexibleConfigUri);
```

Comment:
I could have sworn that I replied to this comment.. Ah well.

By a similar logic, every single object in Context is also vulnerable. And if we do decide to go down this path, this will increase the amount of book-keeping that every class using an FC has to do (imagine the engine - it has to store the URIs of the FCs somewhere, right? What if someone uses that map (or list) to retrieve the receipt and then deletes the FC?

I think a better solution would be to restrict deletion (maybe even access?) of FCs to the creating Module. To be fair, I haven't thought about this *too* hard, so I might be missing some obvious things.

Reply:
It's a fair point you are making about Context objects also being vulnerable. And I don't know how to secure the receipts from reflection if that's what you meant. Is reflection capable of getting absolutely anywhere, including obtaining private variables from objects?

Restricting deletion of FCs might be a better alternative, perhaps with the engine cleaning up by deleting module configs when a game is stopped and the user goes back to the main menu. That been said, some modules, i.e. terrain generators (I think), might be active earlier than a game's startup, and I don't know if there is a way to differentiate between the two. But again, wouldn't reflection make any deletion method ultimately available to any sophisticated enough developer?

Finally, I think restricting access is perhaps overkill. We might theoretically want to restrict a config access so that only dependent modules can access the config of a module they are dependent on. In practice I feel even restricting deletion might be overkill.

So, I'd say let's leave this aspect just for discussion for now. We might revisit it in a later PR.